### ISSUE TITLE
C core hangs on negative http2 tests

### ISSUE THREAD

**Issue by ncteisen** [grpc/grpc#9698]
The c core will hang if it encounters bad http2 frames. For example, if it receives a GOAWAY it will hang rather than complete the RPC.

I believe that this was introduced in #9511. Specifically, the changes to [grpc_chttp2_ack_ping in chttp2_transport.c](https://github.com/grpc/grpc/pull/9511/commits/ad2a11fc3d21765d89b4f86106de9c497dae92c9#diff-ce37d99bf1a5e7e131e56cee8f3f30c6R1250) and [grpc_chttp2_ping_create in frame_ping.c](https://github.com/grpc/grpc/pull/9511/commits/c0118b494ec2df10d7945925e9b1c9da6b868919#diff-6ace7bed4f8f0cd34a8c5c8c1934c17fR43)

To reproduce, start up the negative http2 server with:

```
python test/http2_test/http2_test_server.py
```

Then run the interop client:

```
bins/opt/interop_client  \
  --server_host=127.0.0.1  \
  --server_host_override=localhost  \
  --server_port=8080  \
  --test_case=large_unary  \
```

The test hangs with the message:

```
D0213 16:54:20.584372000 140735237890048 chttp2_transport.c:1302]      Unknown ping response from ipv4:127.0.0.1:8080: 100000000000000
```

It should finish correctly, since the server only sends the GOAWAY after responding to the call.

**Comment by ncteisen** [comment]
cc @makdharma, this is why the Python and C++ negative interop tests are failing with timeouts right now

**Comment by ctiller** [comment]
@ncteisen I'm unable to run this:

```
~/grpc (hang *$%) python test/http2_test/http2_test_server.py 
Traceback (most recent call last):
  File "test/http2_test/http2_test_server.py", line 39, in <module>
    import http2_base_server
  File "/usr/local/google/home/ctiller/grpc/test/http2_test/http2_base_server.py", line 31, in <module>
    import messages_pb2
  File "/usr/local/google/home/ctiller/grpc/test/http2_test/messages_pb2.py", line 23, in <module>
    ms\x18\x01 \x01(\x05\"3\n\rReconnectInfo\x12\x0e\n\x06passed\x18\x01 \x01(\x08\x12\x12\n\nbackoff_ms\x18\x02 \x03(\x05*\x1f\n\x0bPayloadType\x12\x10\n\x0c\x43OMPRESSABLE\x10\x00\x62\x06proto3')
TypeError: __init__() got an unexpected keyword argument 'syntax'
```

**Comment by ncteisen** [comment]
The python server depends on python's protobuf module. This looks like a problem with that. What version of protobuf do you have to python?

**Comment by ncteisen** [comment]
@ctiller, you fixed incidentally fixed this bug with [this commit](https://github.com/grpc/grpc/commit/fd2cd7856b55f391d6bb43d2fd39904b0db3b397) two days after I opened this issue!

I don't know exactly how it fixed it, but I just did a git bisect calling the bugged version good and the magically fixed version bad, and I landed on that commit. Sure enough, checking out the one before it reintroduced the bug.

This unblocks the Python and C++ negative interop tests from going into Jenkins. @makdharma @ericgribkoff
